test/hasPath.js

```diff
@@ -50,4 +59,16 @@ describe('hasPath', function() {
   it('returns false for empty path', function() {
     eq(R.hasPath([], obj), false);
   });
+
+  it('tests paths on non-objects', function() {
```

Comment:
perhaps `it('returns false for non-objects', ...)`

'tests' is very non-specific.

Reply:
yeah we can do that, or we can utilize mochas contexts and [gherkin DSL](http://docs.behat.org/en/v2.5/guides/1.gherkin.html)
```js
context('given second argument is a non-object', function() {
  specify('should return false', function() {
    assertions
  });
});
```
@CrossEye let me know if this is something you like. `context` is alias of `describe`, `specify` of `it`. In this case we have a top level `describe`, we have inner `context` inside `describe` and `specify` inside it.

Reply:
I was only worried about the actual text of the test.  This reads fine to me, but I don't have any real preference between nested `describe`s and this `context`.